ext/intl/listformatter/listformatter_class.c

```diff
@@ -0,0 +1,221 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Bogdan Ungureanu <bogdanungureanu21@gmail.com>              |
+   +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_intl.h"
+#include <unicode/ulistformatter.h>
+#include "listformatter_arginfo.h"
+#include "listformatter_class.h"
+#include "intl_convert.h"
+
+
+static zend_object_handlers listformatter_handlers;
+
+/* {{{ listformatter_free_obj */
+static void listformatter_free_obj(zend_object *object)
+{
+    ListFormatter_object *obj = php_intl_listformatter_fetch_object(object);
+    listformatter_data_free(&obj->lf_data);
+    zend_object_std_dtor(&obj->zo);
+}
+/* }}} */
+
+/* {{{ listformatter_create_object */
+static zend_object *listformatter_create_object(zend_class_entry *class_type)
+{
+    ListFormatter_object *obj;
+    obj = zend_object_alloc(sizeof(ListFormatter_object), class_type);
+    listformatter_data_init(&obj->lf_data);
+    zend_object_std_init(&obj->zo, class_type);
+    object_properties_init(&obj->zo, class_type);
+    obj->zo.handlers = &listformatter_handlers;
+    return &obj->zo;
+}
+/* }}} */
+
+/* {{{ listformatter_create_object */
+PHP_METHOD(IntlListFormatter, __construct)
+{
+    ListFormatter_object *obj = Z_INTL_LISTFORMATTER_P(ZEND_THIS);
+    zend_string *locale;
+    zend_long type = ULISTFMT_TYPE_AND;
+    zend_long width = ULISTFMT_WIDTH_WIDE;
+    ZEND_PARSE_PARAMETERS_START(1, 3)
+        Z_PARAM_STR(locale)
```

Comment:
need to double check locale, e.g. `ZSTR_LEN(locale)` needs to be <= `INTL_MAX_LOCALE_LEN`

Reply:
to expand on that if its len is 0 then locale can fetch a default from intl_locale_get_default(). Might be simpler if locale is just char * then, just an advice.

Reply:
Thanks, makes sense. Done :)